src/rviz/robot/robot_link.cpp

```diff
@@ -668,6 +671,7 @@ void RobotLink::createEntityForGeometryElement(const urdf::LinkConstSharedPtr& l
         sub->setMaterial(mat);
       }
       materials_[sub] = sub->getMaterial();
+      original_materials_[sub] = sub->getMaterial()->clone(sub->getMaterial()->getName() + "_original");
```

Comment:
I don't yet get the point of this source line: Isn't that simply cloning the existing material without changes? How can this result in different behavior?

Reply:
The materials of the line above (Line 673) are changed afterwards in `updateAlpha()` method. See line [405](https://github.com/ros-visualization/rviz/pull/1704/commits/e4dbcdb2eccd3b1669a710e3cdfe87b588dc95d3#diff-34b5917b7e73ed148ec80e571247fbdeca5d564aa6e83d7a24becfbe6544e8dbL405) to 411. Its not that easy to _undo_ these changes when alpha is set back to 1. because then one as to store what scene blending mode and depth writing mode were before for a specific material.